src/serializer/serializer.js

```diff
@@ -1621,6 +1742,17 @@ export class Serializer {
         ),
       ]));
     }
+    if (this.needsVars) {
+      body.push(t.variableDeclaration("var", [
+        t.variableDeclarator(
+          this.varsExpression,
+          t.newExpression(t.identifier("Array"), [t.numericLiteral(this.nextVarsIndex)])
```

Comment:
This is a subtle and important design decision that is not obvious to me. Perhaps some comments somewhere will help future readers understand it.